tt_metal/impl/device/multi_device.cpp

```diff
@@ -134,6 +134,18 @@ int DeviceMesh::num_devices() const
     return mesh_devices.size();
 }
 
+CoreCoord DeviceMesh::compute_with_storage_grid_size() const {
```

Comment:
We use this for sharding from python. We determine shard configs from the grid size, which we are either hardcoded or manually querying from the first device in python.

Reply:
I think if that's the case, you need to loop over all chips in the mesh and assert that they are the same. I feel like this could get very buggy. Or why not just call devices(0).grid_size() from higher level user code?

Reply:
Having `devices(0).grid_size()` from higher level use case is what we have now, but since we share same code between single/multi device you end up scattering code like `device.core_grid if isinstance(device, ttnn.Device) else device.get_devices()[0].core_grid` everywhere, which doesn't seem clean to me as with this user just calls the api and it doesn't matter which type of device object it is.

We can specify in the docs that it is the grid of the first device to make it clear to the user, and if they want they can check all devices manually themselves. Otherwise we have checks if the grid size used doesn't match the per device grid size